### PR TITLE
Increase coverage for lexer string escapes

### PR DESCRIPTION
**Pull request by bungcip**
Coverage Increased
- `src/lexer.rs`: Increased coverage for `unescape_string` error paths.

The added test `test_string_escapes_edge_cases` exercises previously uncovered branches:
1. The `else` block when `\x` is not followed by any hex digits.
2. The overflow check for hex escapes exceeding valid Unicode scalar values.
3. Handling of `\?` and unknown escape sequences.

---
*PR created automatically by Jules for task [2453592095320178419](https://jules.google.com/task/2453592095320178419) started by @bungcip*

### CODE REVIEW

File: src/tests/lexer.rs [hunk 1]
```diff
@@ -290,4 +290,41 @@ mod tests {
             "Unrecognized character should produce Unknown"
         );
     }
+
+    #[test]
+    fn test_string_escapes_edge_cases() {
+        // Test edge cases in string literal unescaping
```
[the remark this answers is not in the document]
Comment (bungcip) [reply src/tests/lexer.rs]: @jules, can you check that?
